app/src/main/java/com/example/healthinspector/Fragments/ProductFinderFragment.java

```diff
@@ -0,0 +1,223 @@
+                @Override
+                public void onResponse(Call call, Response response) throws IOException {
+                    try {
+                        JSONObject jsonResponse = new JSONObject(response.body().string());
+                        Log.i(TAG, jsonResponse.toString());
+                        if(jsonResponse.getJSONArray(Constants.DATA).length() > 0){
+                            locations.get(count).put(Constants.IN_STOCK, true);
+                        }
+                    } catch (JSONException e) {
+                        e.printStackTrace();
+                    }
+                }
+            });
+        }
+        return view;
+    }
+
+    @Override
+    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
+        super.onViewCreated(view, savedInstanceState);
+        //wait for API request to complete, then populate recycler view
+
+        final Handler handler = new Handler();
+        handler.postDelayed(new Runnable() {
+            @Override
+            public void run() {
+                //sort the json array by stores that have the item in stock, then by nearest location
+                Location currentLocation = LocationService.getLastLocation();
+                Collections.sort(locations, new Comparator<JSONObject>() {
+                    @Override
+                    public int compare(JSONObject object1, JSONObject object2) {
+                        Location location1 = new Location("");
+                        Location location2 = new Location("");
+                        try {
+                            location1.setLatitude(object1.getDouble(Constants.LATITUDE));
+                            location1.setLongitude(object1.getDouble(Constants.LONGITUDE));
+
+                            location2.setLatitude(object2.getDouble(Constants.LATITUDE));
+                            location2.setLongitude(object2.getDouble(Constants.LONGITUDE));
+                        } catch (JSONException e) {
+                            e.printStackTrace();
+                        }
+                        if(object1.has(Constants.IN_STOCK) && object2.has(Constants.IN_STOCK)){
+                            //then sort by location
+                            return Float.compare(currentLocation.distanceTo(location1), currentLocation.distanceTo(location2));
+                        }
+                        else if(object1.has(Constants.IN_STOCK) && !object2.has(Constants.IN_STOCK)){
+                            return -1;
+                        }
+                        else if(!object1.has(Constants.IN_STOCK) && object2.has(Constants.IN_STOCK)){
+                            return 1;
+                        }
+                        else{
+                            //just sort by location
+                            return Float.compare(currentLocation.distanceTo(location1), currentLocation.distanceTo(location2));
+                        }
```

Comment:
I like this sorting alogrithm! Another way to do it could be to just by location. and then separately sort by in-stock. This will effectively do the same thing!